gamemodes/terrortown/gamemode/server/sv_armor.lua

```diff
@@ -0,0 +1,184 @@
+end
+
+---
+-- Increases the @{ARMOR} directly about a specific value while keeping the max value.
+-- To add a player's armor(item), @{plymeta:GiveArmor()} should be used!
+-- @param number armor The amount to be increased
+-- @realm server
+function plymeta:IncreaseArmorValue(armor)
+	self:SetArmor(self:GetArmor() + math.max(armor, 0))
+end
+
+---
+-- Decreases the maximum @{ARMOR} directly about a specific value.
+-- @param number armor The amount to be decreased
+-- @realm server
+function plymeta:DecreaseMaxArmorValue(armor)
+	self:SetMaxArmor(self:GetMaxArmor() - math.max(armor, 0))
+end
+
+---
+-- Increases the maximum @{ARMOR} directly about a specific value.
+-- @param number armor The amount to be increased
+-- @realm server
+function plymeta:IncreaseMaxArmorValue(armor)
+	self:SetMaxArmor(self:GetMaxArmor() + math.max(armor, 0))
+end
+
+---
+-- Resets the @{ARMOR} value including the max value, called in @{GM:PlayerSpawn}.
+-- @realm server
+-- @internal
+function plymeta:ResetArmor()
+	self:SetArmor(0)
+	self:SetMaxArmor(0)
+end
+
+---
+-- Sets the initial @{Player} Armor, called in @{GM:TTTBeginRound} after (dumb) players
+-- are respawned. Therefore no armor has to be reset and it is safe that the player
+-- receives their armor.
+-- @realm server
+-- @internal
+function ARMOR:InitPlayerArmor()
+	for _, p in ipairs(player.GetAll()) do
+		if p:IsTerror() then
+			p:GiveArmor(self.cv.armor_on_spawn:GetInt())
+		end
+	end
+end
+
+---
+-- Handles the @{ARMOR} of a @{Player}, called by @{GM:PlayerTakeDamage}
+-- @param Player ply The @{Player} taking damage
+-- @param Entity infl The inflictor
+-- @param Player|Entity att The attacker
+-- @param number amount Amount of damage
+-- @param CTakeDamageInfo dmginfo Damage info
+-- @realm server
+-- @internal
+function ARMOR:HandlePlayerTakeDamage(ply, infl, att, amount, dmginfo)
+	-- fallback for players who prefer the vanilla armor
+	if self.cv.armor_classic:GetBool() and ply:GetArmor() > 0 then
+		-- classic armor only shields from bullet/crowbar damage
+		if dmginfo:IsDamageType(DMG_BULLET) or dmginfo:IsDamageType(DMG_CLUB) then
+			dmginfo:ScaleDamage(0.7)
+		end
+
+		return
+	end
+
+	-- handle different damage type factors, only these four damage types are valid
+	if not dmginfo:IsDamageType(DMG_BULLET) and not dmginfo:IsDamageType(DMG_CLUB)
+		and not dmginfo:IsDamageType(DMG_BURN) and not dmginfo:IsDamageType(DMG_BLAST)
+	then return end
+
+	-- calculate damage
+	local damage = dmginfo:GetDamage()
+	local armor = ply:GetArmor()
+
+	-- normal damage handling when no armor is available
+	if armor == 0 then return end
+
+	self.cv.armor_factor = self.cv.armor_damage_block_pct:GetFloat()
+	self.cv.health_factor = self.cv.armor_damage_health_pct:GetFloat()
+
+	if ply:ArmorIsReinforced() then
+		self.cv.health_factor = self.cv.health_factor - 0.15
+	end
+
+	ply:DecreaseArmorValue(self.cv.armor_factor * damage)
+
+	local new_damage = self.cv.health_factor * damage - math.min(armor, 0)
```

Comment:
`- math.min(armor, 0)` is 0 I think...

Reply:
?

Reply:
see the latest commit, I addressed this issue and added a short explanation in a comment.